File: 1637-widest-vertical-area-between-two-points-containing-no-points/1637-widest-vertical-area-between-two-points-containing-no-points.cpp
class Solution {
public:
    static bool cmp(vector<int>&a, vector<int>&b){
        return a[0] < b[0];
    }
    
    int maxWidthOfVerticalArea(vector<vector<int>>& points) {
        sort(points.begin(), points.end(), cmp);
        
        int res = 0, n = points.size();
        
        for(int i = 1; i < n; i++){
            res = max(res, points[i][0] - points[i - 1][0]);            
        }
        
        return res;
    }
};

File: 1637-widest-vertical-area-between-two-points-containing-no-points/1637-widest-vertical-area-between-two-points-containing-no-points.java
class Solution {
    public int maxWidthOfVerticalArea(int[][] points) {
        Arrays.sort(points, (int[] a, int[] b) -> Integer.compare(a[0], b[0]));

        int res = 0, n = points.length;

        for(int i = 1; i < n; i++){
            res = Math.max(res, points[i][0] - points[i - 1][0]);
        }

        return res;
    }
}